core/src/main/scala/akka/persistence/jdbc/state/scaladsl/DurableStateSequenceActor.scala

```diff
@@ -167,20 +217,78 @@ class DurableStateSequenceActor[A](stateStore: JdbcDurableStateStore[A], config:
     // that happens when we are not yet at the end of the stream
     val isFullBatch = elements.size == batchSize
 
-    if (noGapsFound && isFullBatch) {
-      // Many elements have been retrieved but none are missing
-      // We can query again immediately, as this allows the actor to rapidly retrieve the real max ordering
-      self ! QueryOrderingIds
-      context.become(receive(nextMax, newMissingByCounter, moduloCounter))
+    log.debug(
+      s"findGaps currentMaxOffset [$currentMaxOffset],  elements [${elements.mkString(
+        ", ")}], moduloCounter [$moduloCounter], givenUp [$givenUp] " +
+      s"\n=> nextMax [$nextMax], missingElems [$missingElems], noGapsFound [$noGapsFound], isFullBatch [$isFullBatch]")
+
+    if (noGapsFound) {
+      if (isFullBatch) {
+        // We can query again immediately, as this allows the actor to rapidly retrieve the real max ordering
+        self ! QueryState
+      } else {
+        // keep querying but not immediately
+        scheduleQuery(queryDelay)
+      }
+
+      // cache to be used in next iteration
+      if (elements.nonEmpty) {
+        elements.foreach(e => revisionCache += ((e.pid, e)))
+      }
```

Comment:
@patriknw we don't need this - right ? We are calling the function to populate the cache in the next line ..

Reply:
thanks, it was an accidental leftover